shub/image/list.py

```diff
@@ -52,27 +54,38 @@ def list_cmd_full(target, silent, version):
     version = version or config.get_version()
     image_name = utils.format_image_name(image, version)
     target_conf = config.get_target_conf(target)
-    for spider in list_cmd(image_name,
-                           target_conf.project_id,
-                           target_conf.endpoint,
-                           target_conf.apikey):
+    metadata = list_cmd(image_name,
+                        target_conf.project_id,
+                        target_conf.endpoint,
+                        target_conf.apikey)
+    for spider in metadata.get('spiders', []):
         click.echo(spider)
 
 
 def list_cmd(image_name, project, endpoint, apikey):
     """Short version of list cmd to use with deploy cmd."""
     settings = _get_project_settings(project, endpoint, apikey)
-
-    # Run a local docker container to run list-spiders cmd
-    status_code, logs = _run_list_cmd(project, image_name, settings)
-    if status_code != 0:
+    environment = {'JOB_SETTINGS': json.dumps(settings)}
+    exit_code, logs = _run_cmd_in_docker_container(
+            image_name, 'shub-image-info', environment)
+    if exit_code == 0:
+        return _extract_metadata_from_image_info_output(logs)
+    # shub-image-info command not found, fallback to list-spiders
+    elif exit_code == 127:
+        # FIXME we should pass some value for SCRAPY_PROJECT_ID anyway
+        # to handle `scrapy list` cmd properly via sh_scrapy entrypoint
```

Comment:
(I know this `FIXME` was just copied over, but) What happens when there is no/an empty `SCRAPY_PROJECT_ID` defined, as implemented?